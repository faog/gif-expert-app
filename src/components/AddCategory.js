import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (evt) => {
    setInputValue(evt.target.value);
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories(categories => [inputValue, ...categories]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{ inputValue }</p>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
};
