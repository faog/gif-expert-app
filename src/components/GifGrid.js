import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async() => {
    const url = 'https://api.giphy.com/v1/gifs/search?api_key=SehYrvVMKCk9VRuh9nyEK2w66JLEITpL&q=onepunchman&limit=10';
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map(img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url
      };
    });

    setImages(gifs);
  };

  return (
    <div>
      <h3>{ category }</h3>
      {
        images.map(img =>
          <GifGridItem
            key={img.id}
            {...img}
          />
        )
      }
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string
};
