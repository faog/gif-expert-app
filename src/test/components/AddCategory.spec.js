import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('<AddCategory /> Component', () => {
  let wrapper;
  // const setCategories = () => {};
  const setCategories = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  afterEach(() => {

  });

  test('should be show component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should be change input value', () => {
    const input = wrapper.find('input');
    const value = 'Hola Mundo';
    input.simulate('change', { target: { value } });
    expect(wrapper.find('p').text().trim()).toBe(value);
  });

  test('should not post information with "submit" ', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('should call setCategories and clear text box', () => {
    const value = 'Hola';
    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
