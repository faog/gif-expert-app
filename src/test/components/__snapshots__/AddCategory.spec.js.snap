// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCategory /> Component should be show component 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
